src/read.c

```diff
@@ -2774,10 +2733,6 @@ static void ClearReaderState(void ) {
 static void RestoreReaderState(const struct SavedReaderState *s) {
   memcpy( STATE(ReadJmpError), s->readJmpError, sizeof(syJmp_buf) );
   STATE(UserHasQuit) = s->userHasQuit;
-  ReaderState()->StackNams   = s->stackNams;
-  ReaderState()->ReadTop     = s->readTop;
-  ReaderState()->ReadTilde   = s->readTilde;
-  ReaderState()->CurrLHSGVar = s->currLHSGVar;
```

Comment:
No need to preserve or restore these, as we stack allocate the `ReaderState` for each use freshly on the stack.